### PR TITLE
PSASSET-825-Security Groups with ICMP enabled

### PR DESCRIPTION
**Pull request by rshade**
### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: security/security_groups/icmp_enabled/icmp_enabled.pt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        ds_security_group_rules[i]["self_href"] = ds_security_group_rules[i]["links"][h]["href"]
+      }
+    }
+  }
+  for ( var n = 0; n < ds_networks.length; n++ ){
+    if ( ds_security_group_rules[i]["network_href"] == ds_networks[n]["href"] ){
+      ds_security_group_rules[i]["network_name"] = ds_networks[n]["name"]
+      var network_res=ds_networks[n]["href"].split('/')
+      var sg_res=ds_security_group_rules[i]["security_group_href"].split('/')
+      ds_security_group_rules[i]["server_access_link_root"] = base_url.concat(rs_project_id,'/network_manager#networks/',network_res[3],'/security_groups/',sg_res[5])
```
Comment (arangamani) [inline security/security_groups/icmp_enabled/icmp_enabled.pt]: Why is it called `server_access_link_root`?

File: security/security_groups/icmp_enabled/icmp_enabled.pt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        for ( var l=0; l < ds_security_groups[s]["links"].length; l++ ){
+          if ( ds_security_groups[s]["links"][l]["rel"] == "network" ) {
+            ds_security_group_rules[i]["network_href"] = ds_security_groups[s]["links"][l]["href"]
+          }
+        }
+      } else if (ds_security_group_rules[i]["links"][h]["rel"] == "self") {
+        ds_security_group_rules[i]["self_href"] = ds_security_group_rules[i]["links"][h]["href"]
+      }
+    }
+  }
```
Comment (arangamani) [inline security/security_groups/icmp_enabled/icmp_enabled.pt]: Nitpick: This code is a bit hard to follow. I guess there can be some functions to do common stuffs like extracting the href of a specific link. Not saying we have to change it right away. That can be improved later on.

File: security/security_groups/icmp_enabled/icmp_enabled.pt (hunk 1 above)
Comment (rshade) [reply security/security_groups/icmp_enabled/icmp_enabled.pt]: Because I stole the code from old tag checker.

File: security/security_groups/icmp_enabled/icmp_enabled.pt (hunk 2 above)
Comment (rshade) [reply security/security_groups/icmp_enabled/icmp_enabled.pt]: Definitely and the paths are different. Also some of the items have a right_site style id, others don't. We have to merge right_api 1.5 and 1.6 to get a link in the email on a lot of these. At least security groups and networks use the new ids
